rack-protection/lib/rack/protection/host_authorization.rb

```diff
@@ -8,16 +8,43 @@ module Protection
     ##
     # Prevented attack::   DNS rebinding and other Host header attacks
     # Supported browsers:: all
-    # More infos::         https://en.wikipedia.org/wiki/DNS_rebinding
-    #                      https://portswigger.net/web-security/host-header
+    # More info::          * https://en.wikipedia.org/wiki/DNS_rebinding
+    #                      * https://portswigger.net/web-security/host-header
     #
     # Blocks HTTP requests with an unrecognized hostname in any of the following
-    # HTTP headers: Host, X-Forwarded-Host, Forwarded
+    # HTTP headers: Host, X-Forwarded-Host, Forwarded, based on
+    # {<tt>Request#forwarded_authority</tt>}[https://rubydoc.info/github/rack/rack/Rack/Request/Helpers#forwarded_authority-instance_method] and
```

Comment:
Since the ultimate user of this is likely someone working at the Rack or Sinatra level, I figured it would be helpful to include these specifics about how the hosts are determined beyond just the headers that are used.